fem/transfer.cpp

```diff
@@ -522,6 +522,439 @@ const Operator &L2ProjectionGridTransfer::BackwardOperator()
    return *B;
 }
 
+
+L2ProjectionH1GridTransfer::L2ProjectionH1::L2ProjectionH1(
+   const FiniteElementSpace& fes_ho_, const FiniteElementSpace& fes_lor_)
+   : Operator(fes_lor_.GetVSize(), fes_ho_.GetVSize()),
+     fes_ho(fes_ho_),
+     fes_lor(fes_lor_),
+     P(NULL),
+     p_rtol(1e-13),
+     p_atol(1e-13)
+{
+   Mesh* mesh_ho = fes_ho.GetMesh();
+   Mesh* mesh_lor = fes_lor.GetMesh();
+   int nel_ho = mesh_ho->GetNE();
+   int nel_lor = mesh_lor->GetNE();
+   int ndof_lor = fes_lor.GetNDofs();
+
+   // If the local mesh is empty, skip all computations
+   if (nel_ho == 0) { return; }
+
+   const CoarseFineTransformations& cf_tr = mesh_lor->GetRefinementTransforms();
+
+   int nref_max = 0;
+   Array<Geometry::Type> geoms;
+   mesh_ho->GetGeometries(mesh_ho->Dimension(), geoms);
+   for (int ig = 0; ig < geoms.Size(); ++ig)
+   {
+      Geometry::Type geom = geoms[ig];
+      nref_max = std::max(nref_max, cf_tr.point_matrices[geom].SizeK());
+   }
+
+   // Construct the mapping from HO to LOR
+   // ho2lor.GetRow(iho) will give all the LOR elements contained in iho
+   ho2lor.MakeI(nel_ho);
+   for (int ilor = 0; ilor < nel_lor; ++ilor)
+   {
+      int iho = cf_tr.embeddings[ilor].parent;
+      ho2lor.AddAColumnInRow(iho);
+   }
+   ho2lor.MakeJ();
+   for (int ilor = 0; ilor < nel_lor; ++ilor)
+   {
+      int iho = cf_tr.embeddings[ilor].parent;
+      ho2lor.AddConnection(iho, ilor);
+   }
+   ho2lor.ShiftUpI();
+
+   // ML_inv contains the inverse lumped (row sum) mass matrix
```

Comment:
I know the inversion of a low order lumped mass matrix works for our application. Do we anticipate another application where you would have to or want to invert the whole consistent low order mass matrix? Maybe extend the comment indicating that for now the implementation "chose" to use a lumped mass matrix inversion, unless we think this will extend across all use cases.

Reply:
good point

Reply:
I think the [comment in the header file](https://github.com/mfem/mfem/blob/be70f2a528c34a508b2c5a5bc4b1c983b201b8f4/fem/transfer.hpp#L253) covers this?  If not, I'd be happy to put more details in here.

Reply:
I think adding another comment in the .cpp file doesn't hurt. Just my opinion though.

Reply:
Comments never hurt :)

Reply:
Fixed in latest commit.